helpers/channels/forums/createForumPost.ts

```diff
@@ -0,0 +1,118 @@
+    bot.rest,
+    "post",
+    bot.constants.endpoints.FORUM_START(channelId),
+    {
+      name: options.name,
+      auto_archive_duration: options.autoArchiveDuration,
+      rate_limit_per_user: options.rateLimitPerUser,
+
+      content: options.content,
+      embeds: options.embeds?.map((embed) => bot.transformers.reverse.embed(bot, embed)),
+      allowed_mentions: options.allowedMentions
+        ? {
+          parse: options.allowedMentions?.parse,
+          roles: options.allowedMentions?.roles?.map((id) => id.toString()),
+          users: options.allowedMentions?.users?.map((id) => id.toString()),
+          replied_user: options.allowedMentions?.repliedUser,
+        }
+        : undefined,
+      file: options.file,
+      components: options.components?.map((component) => ({
+        type: component.type,
+        components: component.components.map((subComponent) => {
+          if (subComponent.type === MessageComponentTypes.InputText) {
+            return {
+              type: subComponent.type,
+              style: subComponent.style,
+              custom_id: subComponent.customId,
+              label: subComponent.label,
+              placeholder: subComponent.placeholder,
+              min_length: subComponent.minLength ?? subComponent.required === false ? 0 : subComponent.minLength,
+              max_length: subComponent.maxLength,
+            };
+          }
+
+          if (subComponent.type === MessageComponentTypes.SelectMenu) {
+            return {
+              type: subComponent.type,
+              custom_id: subComponent.customId,
+              placeholder: subComponent.placeholder,
+              min_values: subComponent.minValues,
+              max_values: subComponent.maxValues,
+              options: subComponent.options.map((option) => ({
+                label: option.label,
+                value: option.value,
+                description: option.description,
+                emoji: option.emoji
+                  ? {
+                    id: option.emoji.id?.toString(),
+                    name: option.emoji.name,
+                    animated: option.emoji.animated,
+                  }
+                  : undefined,
+                default: option.default,
+              })),
+            };
+          }
+
+          return {
+            type: subComponent.type,
+            custom_id: subComponent.customId,
+            label: subComponent.label,
+            style: subComponent.style,
+            emoji: "emoji" in subComponent && subComponent.emoji
+              ? {
+                id: subComponent.emoji.id?.toString(),
+                name: subComponent.emoji.name,
+                animated: subComponent.emoji.animated,
+              }
+              : undefined,
+            url: "url" in subComponent ? subComponent.url : undefined,
+            disabled: "disabled" in subComponent ? subComponent.disabled : undefined,
```

Comment:
```suggestion
            url: subComponent.url ?? undefined,
            disabled: subComponent.disabled ?? undefined,
```

Reply:
note: I just want to know why "in", consider waiting for ITOH/Skill4Killz review before changing this